test/extended/machines/cluster.go

```diff
@@ -100,4 +119,293 @@ var _ = g.Describe("[sig-node] Managed cluster", func() {
+	command := "exec chroot /host journalctl --list-boots"
+	isTrue := true
+	zero := int64(0)
+	ctx := context.Background()
+	name := fmt.Sprintf("list-boots-%s-%d", nodeName, attempt)
+	pod, err := kubeClient.CoreV1().Pods(namespace).Create(context.Background(), &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.PodSpec{
+			Tolerations: []corev1.Toleration{
+				{
+					Effect:   "NoSchedule",
+					Key:      "node-role.kubernetes.io/master",
+					Operator: corev1.TolerationOpExists,
+				},
+			},
+			HostPID:            true,
+			RestartPolicy:      corev1.RestartPolicyNever,
+			NodeName:           nodeName,
+			ServiceAccountName: serviceAccountName,
+			Volumes: []corev1.Volume{
+				{
+					Name: "host",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/",
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name: "list-boots",
+					SecurityContext: &corev1.SecurityContext{
+						RunAsUser:  &zero,
+						Privileged: &isTrue,
+					},
+					Image: image.ShellImage(),
+					Command: []string{
+						"/bin/bash",
+						"-c",
+						command,
+					},
+					TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/host",
+							Name:      "host",
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if apierrors.IsAlreadyExists(err) {
+		return getNumberOfBootsForNode(kubeClient, nodeName, attempt+1)
+	}
+	if err != nil {
+		return 0, fmt.Errorf("failed to create pod %s: %v", name, err)
+	}
+
+	podName := pod.Name
+
+	// Wait for pod to run
+	err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+		podGet, getErr := kubeClient.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+		if getErr != nil {
+			return false, getErr
+		}
+		switch podGet.Status.Phase {
+		case corev1.PodSucceeded:
+			return true, nil
+		case corev1.PodFailed:
+			return true, fmt.Errorf("journalctl command in pod %s failed. Pod status: %#v", name, podGet.Status)
+		default:
+			return false, nil
+		}
+	})
+	if err != nil {
+		return 0, fmt.Errorf("pod %s failed to complete: %v", name, err)
+	}
+
+	// Fetch pod logs
+	linesInPodLogs := -1
+	podLogOpts := corev1.PodLogOptions{}
+	req := kubeClient.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream(ctx)
+	if err != nil {
+		return linesInPodLogs, fmt.Errorf("failed to open stream to read %s pod logs: %v", name, err)
+	}
+	defer podLogs.Close()
+
+	// Count number of boots
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return linesInPodLogs, fmt.Errorf("failed to copy information from podLogs to buf: %v", err)
+	}
+	linesInPodLogs = strings.Count(buf.String(), "\n") - 1
+	if err != nil {
+		return linesInPodLogs, fmt.Errorf("failed to fetch boot list from %s pod logs: %v", name, err)
```

Comment:
If the line count return value is valid when a non-nil error is also returned, it's surprising that it can be a negative number. I think it would be clearer if the calling code were to omit the map entry for nodes that hit an error.

Reply:
Good idea, update this to:
* return 0 if error occurred
* include pod logs in the error message
* throw error of number of boots is less than 1